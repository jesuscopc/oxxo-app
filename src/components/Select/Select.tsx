import React from 'react';

export type TCatalog = {
  id: string;
  name: string;
};

interface TSelect {
  label: string;
  options: TCatalog[];
  onSelect: React.ChangeEventHandler<HTMLSelectElement>;
  name: string;
  isLoading?: boolean;
}

const Select = ({ label, name, options, isLoading, onSelect }: TSelect): React.ReactElement => {
  return (
    <>
      <label className="font-black pb-2">{label}</label>
      <select
        className="p-5 mb-3 shadow-lg bg-blue-400 text-white rounded-lg font-bold"
        onChange={onSelect}
        name={name}
      >
        {isLoading ? (
          <option>Loading...</option>
        ) : (
          options.map(({ id, name }) => (
            <option className="appearance-none h-3  " key={id} value={id}>
              {name}
            </option>
          ))
        )}
      </select>
    </>
  );
};

export default Select;
