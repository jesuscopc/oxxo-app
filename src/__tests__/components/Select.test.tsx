import * as React from 'react';
import { shallow } from 'enzyme';
import Select from 'components/Select/Select';

describe('SUit test Select', () => {
  test('should be render correctly', () => {
    const wrapper = shallow(
      <Select
        label="Selecciona Pais"
        name="pais"
        options={[{ id: '1', name: 'option1' }]}
        isLoading={false}
        onSelect={() => {}}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should be show loading legend', () => {
    const wrapper = shallow(
      <Select
        label="Selecciona Pais"
        name="pais"
        options={[{ id: '1', name: 'option1' }]}
        isLoading={true}
        onSelect={() => {}}
      />
    );
    expect(wrapper.find('option').text()).toBe('Loading...');
  });

  // test('should be submit form and bubble keyword', () => {
  //   const wrapper = shallow(<SearchBox handleSubmit={() => {}} loading={true} />);
  //   wrapper.find('form').simulate('submit', { preventDefault: () => {} });
  //   expect(wrapper.find('button')).toEqual({});
  // });
});
