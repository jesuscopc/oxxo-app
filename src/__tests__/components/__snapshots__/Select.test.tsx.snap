// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SUit test Select should be render correctly 1`] = `
<Fragment>
  <label
    className="font-black pb-2"
  >
    Selecciona Pais
  </label>
  <select
    className="p-5 mb-3 shadow-lg bg-blue-400 text-white rounded-lg font-bold"
    name="pais"
    onChange={[Function]}
  >
    <option
      className="appearance-none h-3  "
      key="1"
      value="1"
    >
      option1
    </option>
  </select>
</Fragment>
`;
